tests/test_segment.py

```diff
@@ -67,6 +67,11 @@ def __unit_test_structure_function(metric):
         # And that the metric is 0
         assert np.allclose(score, 0)
 
+    # Test for non-matching numbers of intervals and labels
```

Comment:
Haha, just noticed this - was this to try to maintain coverage?  Happy to include this but it's a bit off-topic.